tpls/appcache-frame.tpl

```diff
@@ -90,9 +90,6 @@
         clearInterval(downloadingInterval);
         downloadingInterval = null;
       }
-
-      applicationCache.removeEventListener('downloading', onDownloadingEvent);
-      applicationCache.removeEventListener('progress', onDownloadingEvent);
```

Comment:
Isn't it by design (and desired) that `progress` happens multiple times? Even with the current code those events come in the same way in the first 5 seconds. I don't see the problem, sorry.

What about just keeping the `updateready` event? After all that's the single most important one to get updates working instantly AFAIK. That would still remain a simple fix that could be released soon.

As for a more complete fix I think it would be best if the event *handlers* (perhaps outside of the iframe) can deal with various event "frequencies". After all we learned that browser behaviour is somewhat unpredictable and nobody can guarantee what events come in in what order...